webapp/src/pages/Settings.tsx

```diff
@@ -1,78 +1,692 @@
+import { Warning } from "@mui/icons-material";
 import {
   Box,
   Button,
   Checkbox,
-  Container,
-  FormControl,
+  Divider,
   FormControlLabel,
-  FormGroup,
   FormHelperText,
+  InputBaseComponentProps,
+  Paper,
+  TextField,
+  Tooltip,
+  Typography,
 } from "@mui/material";
+import AccordionLayout from "components/AccordionLayout";
+import _ from "lodash";
 import React from "react";
 import { useParams } from "react-router-dom";
 import { getConfigEndpoint, updateConfigEndpoint } from "services/api";
-import { AzimuthConfig } from "types/api";
+import { AzimuthConfig, PipelineDefinition } from "types/api";
+import { PickByValue } from "types/models";
+
+const STEPPER: Record<string, InputBaseComponentProps> = {
+  iterations: { min: 0, max: 100, step: 1 },
+  high_epistemic_threshold: { min: 0, max: 1, step: 0.1 },
+  conflicting_neighbors_threshold: { min: 0, max: 1, step: 0.1 },
+  no_close_threshold: { min: 0, max: 1, step: 0.1 },
+  min_num_per_class: { min: 0, max: 100, step: 10 },
+  max_delta_class_imbalance: { min: 0, max: 1, step: 0.1 },
+  max_delta_representation: { min: 0, max: 0.1, step: 0.01 },
+  max_delta_mean_tokens: { min: 0.0, max: 10.0, step: 1.0 },
+  max_delta_std_tokens: { min: 0.0, max: 10.0, step: 1.0 },
+  short_sentence_max_token: { min: 0, max: 100, step: 1 },
+  long_sentence_min_token: { min: 0, max: 100, step: 1 },
```

Comment:
I would
- set `min_num_per_class`'s `step` to `1` instead of `10`;
- set `max_delta_representation`'s `max` to `1` as it's a percentage and `max: 0.1` seems arbitrary;
- remove the `max`es that are not `1`, as they seem arbitrary;
- set a decimal `step` to all `float` fields, including `max_delta_mean_tokens` and `max_delta_std_tokens` to `0.1`
- remove `step: 1` as it is the default; and
- remove the unnecessary `.0`s;
```suggestion
  iterations: { min: 0 },
  high_epistemic_threshold: { min: 0, max: 1, step: 0.1 },
  conflicting_neighbors_threshold: { min: 0, max: 1, step: 0.1 },
  no_close_threshold: { min: 0, max: 1, step: 0.1 },
  min_num_per_class: { min: 0 },
  max_delta_class_imbalance: { min: 0, max: 1, step: 0.1 },
  max_delta_representation: { min: 0, max: 1, step: 0.01 },
  max_delta_mean_tokens: { min: 0, step: 0.1 },
  max_delta_std_tokens: { min: 0, step: 0.1 },
  short_sentence_max_token: { min: 0 },
  long_sentence_min_token: { min: 0 },
```
@gabegma, any comments on the values? Like steps of `0.01` vs `0.1` for example.

Reply:
All your suggestions LGTM!
The steps are for the arrows beside the box where the user can type, right? So the user could still type `0.15` and have a value in between the steps increment?

Reply:
Thanks! That is correct.